homeassistant/components/nam/config_flow.py

```diff
@@ -87,38 +144,57 @@ async def async_step_zeroconf(
         await self.async_set_unique_id(format_mac(mac))
         self._abort_if_unique_id_configured({CONF_HOST: self.host})
 
-        self.context["title_placeholders"] = {
-            ATTR_NAME: discovery_info[ATTR_NAME].split(".")[0]
-        }
-
         return await self.async_step_confirm_discovery()
 
     async def async_step_confirm_discovery(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle discovery confirm."""
-        errors: dict = {}
+        errors: dict[str, str] = {}
 
         if user_input is not None:
             return self.async_create_entry(
-                title=cast(str, self.host),
+                title=self.host,
                 data={CONF_HOST: self.host},
             )
 
         self._set_confirm_only()
 
         return self.async_show_form(
             step_id="confirm_discovery",
-            description_placeholders={CONF_HOST: self.host},
+            description_placeholders={"host": self.host},
             errors=errors,
         )
 
-    async def _async_get_mac(self, host: str) -> str:
-        """Get device MAC address."""
-        websession = async_get_clientsession(self.hass)
-        nam = NettigoAirMonitor(websession, host)
-        # Device firmware uses synchronous code and doesn't respond to http queries
-        # when reading data from sensors. The nettigo-air-monitor library tries to get
-        # the data 4 times, so we use a longer than usual timeout here.
-        async with async_timeout.timeout(30):
-            return await nam.async_get_mac_address()
+    async def async_step_reauth(self, data: dict[str, Any]) -> FlowResult:
+        """Handle configuration by re-auth."""
+        if entry := self.hass.config_entries.async_get_entry(self.context["entry_id"]):
+            self.entry = entry
+        self.host = data[CONF_HOST]
+        self.context["title_placeholders"] = {"host": self.host}
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            try:
+                await async_get_mac(self.hass, self.host, user_input)
+            except (ApiError, AuthFailed, ClientConnectorError, asyncio.TimeoutError):
+                return self.async_abort(reason="reauth_unsuccessful")
```

Comment:
Why don't we should an error in the form to the user and let them try again? Maybe they entered the wrong password?

Reply:
Good point. I'll change this.